server/src/services/watcherService.js

```diff
@@ -1,20 +1,22 @@
 import blockSyncService from "./blockSyncService.js";
 import eventService from "./eventService.js";
-import { canvasDeployerAbi, chainsConfig } from "../config.js";
 import { createPublicClient, http, webSocket } from "viem";
+import { CANVAS_DEPLOYER_ABI } from "../constants/abis.js";
+import chains from "../constants/chains.js";
+import { CANVAS_DEPLOYERS } from "../constants/contractAddresses.js";
 
 // Helper function to create an HTTP client for a given chain
-const createHttpClient = (chain, rpcUrl) =>
+const createHttpClient = (chain) =>
   createPublicClient({
     chain,
-    transport: http(rpcUrl),
+    transport: http(chain.rpcUrls.custom?.http[0]),
```

Comment:
```suggestion
    transport: http(chain.rpcUrls.custom?.http[0]),
```
what will happend if `chain.rpcUrls.custom` is `undefined`?